src/components/ui/TagPicker/TagPicker.tsx

```diff
@@ -0,0 +1,180 @@
+import React, {createRef, useEffect, useState} from "react";
+import styles from "./TagPicker.module.scss"
+import {Tag} from "../../../common/models/Tag";
+import Close from "../../../assets/vectors/close.svg"
+import Add from "../../../assets/vectors/add.svg"
+import cs from "classnames";
+
+interface TagPickerProps {
+    tags: Tag[]
+    value: Tag[]
+    maxLength?: number
+    onChange: (tags: Tag[]) => void
+    onCreate: (newTagName: string) => void
+}
+
+interface TagComponentProps {
+    label: string
+    onRemoveClick: () => void
+}
+
+const TagComponent: React.FC<TagComponentProps> = ({label, onRemoveClick}) => (
+    <div className={styles.tag}>
+        {label}
+        <Close className={cs(styles.close, styles.pointer)} onClick={onRemoveClick}/>
+    </div>
+)
```

Comment:
move this to separate file
react is about splitting big things into small things